linode_metadata/metadata_client.py

```diff
@@ -110,6 +112,38 @@ def check_connection(self):
                 "Can't access Metadata service. Please verify that you are inside a Linode."
             ) from e
 
+    def check_token(self):
```

Comment:
Would it make sense to make this method private? I'm not sure if users would use this as opposed to directly calling `MetadataClient.refresh_token()`.